marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -138,13 +138,13 @@ datumWorker Coordinator{_barrier, _channel} path = do
 utxoWorker_
   :: (Utxo.UtxoIndexer -> IO ())  -- ^ callback function used in the queryApi thread, needs to be non-blocking
   -> Utxo.Depth
-  -> Maybe TargetAddresses                       -- ^ Target addresses to filter for
+  -> Maybe TargetAddresses        -- ^ Target addresses to filter for
   -> Coordinator
   -> TChan (ChainSyncEvent (BlockInMode CardanoMode))
   -> FilePath
   -> IO (IO (), MVar Utxo.UtxoIndexer)
 utxoWorker_ callback depth maybeTargetAddresses Coordinator{_barrier} ch path = do
-  ix <- Utxo.open path depth
+  ix <- Utxo.open path depth True -- open SQLite with depth=depth and perform SQLite vacuum
```

Comment:
Either add a parameter for `vacuum` or set it to `False`. It's buggy on rollback if our depth is too low.

Reply:
I added TODO item to allow user decide to vacuum or not. And yes, at low depth, this is an issue and the tests clearly show that

Reply:
Is it that bad performance-wise to disable it by default at the moment?
If we keep it, we may have to add a hack to tests that deal with rollbacks until we find a solution, and it's easy to forget to remove the hack once it's fixed.

Reply:
Sorry, I didn't see the previous comment. :)